core/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -148,7 +149,7 @@ public void onNoLongerMaster(String source) {
                     @Override
                     public void onFailure(String source, Exception e) {
                         //if the reroute fails we only log
-                        logger.debug("failed to perform [{}]", e, source);
+                        logger.debug(new ParameterizedMessage("failed to perform [{}]", source), e);
```

Comment:
Is this ParameterizedMessage something required for log4j2? It means we are constructing a new object on every logging call, regardless of whether the log message will be written (yes it is not as expensive as a string construction, but it is still doing allocation, and the whole point of parameterized log methods were to minimize work when no logging should be performed for the given level).